packages/dds/migration-shim/src/migrationDeltaHandler.ts

```diff
@@ -17,43 +17,53 @@ import { type IShimDeltaHandler } from "./types";
  * check for the v2 stamp.
  */
 export class MigrationShimDeltaHandler implements IShimDeltaHandler {
-	private oldHandler?: IDeltaHandler;
-	private newHandler?: IDeltaHandler;
+	private legacyTreeHandler?: IDeltaHandler;
+	private newTreeHandler?: IDeltaHandler;
 	public constructor(
-		// This is a hack, maybe parent handler would be better
+		// Maybe it would be better to pass in a different interface?
 		public readonly processMigrateOp: (
 			message: ISequencedDocumentMessage,
 			local: boolean,
 			localOpMetadata: unknown,
 		) => boolean,
 	) {}
-	private get handler(): IDeltaHandler {
-		const handler = this.newHandler ?? this.oldHandler;
+
+	public stampV2Ops(messageContent: unknown): void {
```

Comment:
This will sometimes need to stamp v1 instead

Reply:
I'm actually thinking we can drop v1 ops simply by checking if they have the v2 stamp or not. We may not need to add the v1 stamp. The migrate op also will not have the v2 stamp, so it will by default be dropped as well. I'll add a comment.